perception/object_recognition/detection/lidar_apollo_instance_segmentation/src/node.cpp

```diff
@@ -17,19 +17,25 @@
 #include "lidar_apollo_instance_segmentation/node.h"
 #include "lidar_apollo_instance_segmentation/detector.h"
 
-LidarInstanceSegmentationNode::LidarInstanceSegmentationNode() : nh_(""), pnh_("~")
+LidarInstanceSegmentationNode::LidarInstanceSegmentationNode() : Node("lidar_apollo_instance_segmentation_node")
 {
-  detector_ptr_ = std::make_shared<LidarApolloInstanceSegmentation>();
+}
+
+void LidarInstanceSegmentationNode::init()
+{
+  using std::placeholders::_1;
+  detector_ptr_ = std::make_shared<LidarApolloInstanceSegmentation>(this->shared_from_this());
```

Comment:
Why not use a raw pointer for `LidarApolloInstanceSegmentation` here? Since  `LidarApolloInstanceSegmentation` is owned by the `LidarInstanceSegmentationNode`, their life times are bounded together anyway. 

The advantage would be to move this logic to the constructor. Also this way, `LidarApolloInstanceSegmentation` can be used in a node that is not meant to spin safely since we wouldn't be assuming that someone created a `shared_ptr` to it. 

One possible alternative/compromise could be passing a fake `shared_ptr` to the `LidarApolloInstanceSegmentation`  like  `std::shared_ptr<LidarInstanceSegmentationNode>(this, [](){});` so that `LidarApolloInstanceSegmentation` still keeps smart pointer interfaces and only in `LidarInstanceSegmentationNode` we would be abusing the API since we already know the life time relation of both objects.

Though I won't insist too much since assumption of the existence of a `shared_ptr` to nodes seems quite widespread in ROS2 codebase in general from what i gather, so if you feel this is more idiomatic, we can keep it.

Reply:
fixed in https://github.com/tier4/Pilot.Auto/pull/98/commits/d50e3d6ba83733b35b5d861b85994d1a87868733